src/d3a/d3a_core/util.py

```diff
@@ -472,3 +471,12 @@ def is_external_matching_enabled():
     """
     return (ConstSettings.IAASettings.BID_OFFER_MATCH_TYPE ==
             BidOfferMatchAlgoEnum.EXTERNAL.value)
+
+
+def is_time_slot_in_past_markets(time_slot: DateTime, current_time_slot: DateTime):
+    """Checks if the time_slot should be in the area.past_markets."""
```

Comment:
Nit: I think this function might be refactored a bit for clarity. Given the method name, if the function returns `True` I would expect the `time_slot` to be _already_ in `area.past_markets`. Instead, we are actually checking whether `time_slot` _should be_ in `area.past_markets` (but it's probably not there yet).

Maybe a different naming would help (e.g. `is_market_slot_expired` or similar). What do you think?

Reply:
Still regarding this function: I think it would be cool to have classes for it. I am thinking about an interface like the following (as an example):
```python
expiration_timer = SettlementMarketExpirationTimer() if ... else DefaultExpirationTimer()
# Then in the strategy you would have
state = LoadState(market_expiration_timer=expiration_timer)
```
Just an idea, not necessary for this PR.

Reply:
I would vote for `is_timeslot_before_spot_and_settlement_markets`  but it is overly verbose

Reply:
> Still regarding this function: I think it would be cool to have classes for it. I am thinking about an interface like the following (as an example):
> 
> ```python
> expiration_timer = SettlementMarketExpirationTimer() if ... else DefaultExpirationTimer()
> # Then in the strategy you would have
> self.state = LoadState(market_expiration_timer=expiration_timer)
> # And in the State methods
> if self.expiration_timer.is_expired(time_slot):
>     do_something()
> ```
> 
> This would unify the state interface for all kinds of markets (for the expiration checks, at least).
> Just an idea, not necessary for this PR.

This looks like a lot of Boilerplate code for a simple comparison. Not sure if the proposed `SettlementMarketExpirationTimer ` will be used in the future.

Reply:
`is_time_slot_in_past_markets` occurs 8 times in 3 files. I think that in both cases we would have boilerplate code that would benefit from further refactoring :)

Reply:
:-) Indeed, I would like to skip this for the sake of time though.